seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/batch/FlinkBatchExecution.java

```diff
@@ -118,6 +118,6 @@ public Config getConfig() {
     }
 
     private boolean whetherExecute(List<FlinkBatchSink> sinks) {
-        return sinks.stream().anyMatch(s -> !"ConsoleSink".equals(s.getPluginName()));
+        return sinks.stream().anyMatch(s -> !"ConsoleSink".equals(s.getPluginName()) && !"AssertSink".equals(s.getPluginName()));
```

Comment:
In my knowledge, you cannot add this check here, this will not call execute method, so your flink job will not be submitted.